### PR TITLE
tests: add jammy to spread executions

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This change includes new ubuntu Jammy as part of the regular spread executions.

### CODE REVIEW

File: tests/lib/external/snapd-testing-tools/.github/workflows/tests.yaml [hunk 1]
```diff
@@ -1,8 +1,11 @@
 name: Jobs
 on:
+  pull_request:
```
Comment (stolowski) [inline tests/lib/external/snapd-testing-tools/.github/workflows/tests.yaml:3]: Is this change intended for this PR?

File: tests/lib/external/snapd-testing-tools/.github/workflows/tests.yaml [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
+    branches:
+      - main
   push:
     branches:
-      - '**'        # matches every branch
+      - main
```
Comment (stolowski) [inline tests/lib/external/snapd-testing-tools/.github/workflows/tests.yaml:8]: Same.

File: tests/lib/external/snapd-testing-tools/README.md [hunk 3]
```diff
@@ -33,6 +33,19 @@ chat in realtime with the snapd team and our wider community on the
...
+
+Read the following considerations before adding new tools:
+
+ - Each tool needs to be accompanied by at least 1 spread test in `tests/<tool-name>/`
+ - At least 1 spread test needs to be included in the tests directory for each tool
+ - If the tool is a shell script, it needs to first pass a [ShellCheck](https://github.com/koalaman/shellcheck) assessment
+ - All tools need to be as generic as possible
+ - Each tool must also provide a command line interface (CLI), including _help_ output
+
+
```
Comment (stolowski) [inline tests/lib/external/snapd-testing-tools/README.md:48]: Same.

**Review by stolowski — COMMENTED** [review]
Looks good, but a few comments about changes that look unrelated.

File: tests/lib/external/snapd-testing-tools/.github/workflows/tests.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply tests/lib/external/snapd-testing-tools/.github/workflows/tests.yaml:3]: @stolowski thanks for the review, I had to import snapd-testing-tools to support jammy and when I do that It includes all the changes in that project since the last import.

When we do the import we import the main branch by doing:
# git subtree pull --prefix tests/lib/external/snapd-testing-tools/ https://github.com/snapcore/snapd-testing-tools.git main --squash

**Review by stolowski — APPROVED** [review]
LGTM

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -276,6 +276,7 @@ jobs:
         - ubuntu-20.04-64
         - ubuntu-21.04-64
```
Comment (bboozzoo) [inline .github/workflows/test.yaml:277]: Can we drop 21.04 in a followup then?

File: tests/main/cgroup-devices-v1/task.yaml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 summary: measuring basic properties of device cgroup
 
 # Disable the test on all systems that boot with cgroup v2
-systems: [ -fedora-33-*, -fedora-34-*, -fedora-35-*, -debian-11-*, -debian-sid-*, -arch-*, -opensuse-tumbleweed-*, -ubuntu-21.10-*]
+systems: [ -fedora-33-*, -fedora-34-*, -fedora-35-*, -debian-11-*, -debian-sid-*, -arch-*, -opensuse-tumbleweed-*, -ubuntu-21.10-*, -ubuntu-22.04-*]
```
Comment (bboozzoo) [inline tests/main/cgroup-devices-v1/task.yaml:4]: Reminds me we should probably clean up fedora 33 at some point.

**Review by bboozzoo — APPROVED** [review]
Thanks

File: .github/workflows/test.yaml (hunk 4 above)
Comment (mvo5) [reply .github/workflows/test.yaml:277]: Yes please

**Comment by mvo5** [comment]
The tests fail right now with:
```
2022-01-20 18:36:56 Cannot allocate google:ubuntu-22.04-64: cannot find any Google image matching "ubuntu-os-cloud-devel/ubuntu-2204-lts" on project "ubuntu-os-cloud-devel"
```

File: .github/workflows/test.yaml (hunk 4 above)
Comment (sergiocazzolato) [reply .github/workflows/test.yaml:277]: yes

**Comment by sergiocazzolato** [comment]
@mvo5, thanks for the review, the tests are failing because the image ubuntu-os-cloud-devel/ubuntu-2204-lts is in the second page and as spread doesn't have pagination for google backend, it can't get the image. I tested the image by using a local fork of spread but we need the pagination to get that image from the ubuntu-os-cloud-devel project.
